### ISSUE TITLE
Replace deprecated C++ functions

### ISSUE THREAD

**Issue by hadley** [r-dbi/odbc#676]
```
     nanodbc/nanodbc.cpp:301:32: warning: 'codecvt_utf8_utf16<char16_t>' is deprecated [-Wdeprecated-declarations]
     nanodbc/nanodbc.cpp:301:16: warning: 'wstring_convert<std::codecvt_utf8_utf16<char16_t>, char16_t>' is deprecated [-Wdeprecated-declarations]
```

Due to https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2017/p0618r0.html

**Comment by hadley** [comment]
I'll try and get through the current submission with this but we'll need to fix for the next release.

**Comment by detule** [comment]
For what is worth, `nanodbc` is waiting for the standard to provide an alternative:

https://github.com/nanodbc/nanodbc/issues/74

**Comment by hadley** [comment]
In that case maybe we should pragma the warnings away.

**Comment by hadley** [comment]
Fixed in #688
